### PR TITLE
feat(actions): add `go-mod-tidy` and `go-mod-tidy-vendor` plugins

### PR DESCRIPTION
**Pull request by sheldonhull**
- `go-mod-tidy` for enabling action on file change for `go.mod`.
- `go-mod-tidy-vendor` for enabling action on file change for `go.mod` that also runs go mod vendor afterward. This helps improve vendoring workflow to consistent and avoid inconsistent vendoring issues by forgetting.

executable bit is set on both scripts.

### CODE REVIEW

File: actions/go-mod-tidy/plugin.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+version: 0.1
+actions:
+  definitions:
+    - id: go-mod-tidy
+      display_name: Go Mod Tidy
+      description: Runs go mod tidy when changes are detected to go.mod.
+      run: ./go-mod-tidy-vendor.sh
```
Comment (sxlijin) [inline actions/go-mod-tidy/plugin.yaml:7]:
```suggestion
      run: go mod tidy
```

File: actions/go-mod-tidy-vendor/plugin.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+version: 0.1
+actions:
+  definitions:
+    - id: go-mod-tidy-vendor
+      display_name: Go Mod Tidy & Vendor
+      description: Runs go mod tidy followed by go mod vendor.
+      run: ./go-mod-tidy-vendor.sh
+      import_to_global: true
```
Comment (sxlijin) [inline actions/go-mod-tidy-vendor/plugin.yaml:8]: I don't think this field exists on action definitions, but we also don't have good CI checks on PRs into this repo yet.

**Review by sxlijin — COMMENTED** [review]
What do you think about specifying `runtime: go` on these actions?

IIRC for your use case you use your system runtime for most things.

**Comment by sheldonhull** [comment]
You can't chain go mod vendor after tidy with go runtime. Go mod tidy could be though. Want to me to adjust that one?

File: actions/go-mod-tidy-vendor/plugin.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+version: 0.1
+actions:
+  definitions:
+    - id: go-mod-tidy-vendor
+      display_name: Go Mod Tidy & Vendor
+      description: Runs go mod tidy followed by go mod vendor.
+      run: ./go-mod-tidy-vendor.sh
```
Comment (sxlijin) [inline actions/go-mod-tidy-vendor/plugin.yaml:7]:
```suggestion
      run: ${plugin}/actions/go-mod-tidy-vendor/go-mod-tidy-vendor.sh
```

**Comment by sxlijin** [comment]
> You can't chain go mod vendor after tidy with go runtime. Go mod tidy could be though. Want to me to adjust that one?

Yeah, let's make that change, and then I can go and test this change manually.

(You can do that yourself by specifying this as a local plugin, but there's a bug in how plugin configs work right now. We've fixed it but haven't released the fix yet.)

File: actions/go-mod-tidy/plugin.yaml [hunk 4]
```diff
@@ -4,7 +4,7 @@ actions:
     - id: go-mod-tidy
       display_name: Go Mod Tidy
       description: Runs go mod tidy when changes are detected to go.mod.
-      run: ./go-mod-tidy-vendor.sh
-      import_to_global: true
+      runtime: go
+      run: mod tidy
```
Comment (sxlijin) [inline actions/go-mod-tidy/plugin.yaml:8]:
```suggestion
      run: go mod tidy
```

Comment (dapirian) [reply actions/go-mod-tidy/plugin.yaml:8]: I think you still need this change for it to work

Comment (EliSchleifer) [reply actions/go-mod-tidy/plugin.yaml:8]: I think the intention here is to call go-mod-tidy.sh 

So probably want the run to be 
go-mod-tidy.sh

Comment (EliSchleifer) [reply actions/go-mod-tidy/plugin.yaml]: Or this really
      run: ${plugin}/actions/go-mod-tidy/go-mod-tidy.sh

**Review by dapirian — APPROVED** [review]

File: actions/go-mod-tidy-vendor/plugin.yaml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+version: 0.1
+actions:
+  definitions:
+    - id: go-mod-tidy-vendor
+      display_name: Go Mod Tidy & Vendor
+      description: Runs go mod tidy followed by go mod vendor.
```
Comment (dapirian) [inline actions/go-mod-tidy-vendor/plugin.yaml:6]:
```suggestion
      description: Runs go mod tidy followed by go mod vendor.
      runtime: go
```

**Review by dapirian — APPROVED** [review]

**Pull request merged**